docs/types.rst

```diff
@@ -337,7 +337,8 @@ function type should not return anything, the whole ``returns (<return types>)``
 part has to be omitted.
 
 By default, function types are internal, so the ``internal`` keyword can be
-omitted.
+omitted. Note that this is not the case for functions defined on a contract,
```

Comment:
So I've reread this part of the documentation again, and I agree this isn't the right way to explain this at all. I'm going to close this PR.